### PR TITLE
Add lesson modal w/ drag n drop functionaliy

### PR DESCRIPTION
**Pull request by rrrliu**
Clicking on add new lesson opens up a modal where teachers can upload pdfs.

### CODE REVIEW

**Comment by bchee** [comment]
Looks pretty good. Can you post screenshots of what the modal looks like?

File: client/package.json [hunk 1]
```diff
@@ -21,8 +21,11 @@
     "react-chart-histogram": "^0.2.1",
     "react-chartkick": "^0.3.0",
     "react-dom": "^16.5.2",
+    "react-dropzone": "^7.0.1",
+    "react-modal": "^3.6.1",
     "react-router-dom": "^4.3.1",
-    "react-scripts": "1.1.5"
+    "react-scripts": "1.1.5",
+    "superagent": "^4.0.0-beta.5"
```
[the remark this answers is not in the document]
Comment (duancaro) [reply client/package.json]: :(

File: client/src/components/Lessons/AddLessonModal/AddLessonModal.jsx [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  }
+
+  render() {
+    const { handleCloseModal } = this.props;
+
+    return (
+      <div className="lesson-modal">
+        <div className="top-bar top-modal" />
+        <form className="contents">
+          <h1 className="bad-code">Add New lesson</h1>
```
[the remark this answers is not in the document]
Comment (duancaro) [reply client/src/components/Lessons/AddLessonModal/AddLessonModal.jsx]: done

File: client/src/components/Lessons/Lessons.css [hunk 3]
```diff
@@ -25,3 +25,35 @@
...
+.Overlay {
+  position: fixed;
+  background-color: rgba(0, 0, 0, 0.3);
+  height: 100%;
+  width: 100%;
+  top: 0;
+  bottom: 0;
+}
+
+/*
```
[the remark this answers is not in the document]
Comment (duancaro) [reply client/src/components/Lessons/Lessons.css]: done

File: client/src/components/Lessons/Lessons.jsx [hunk 4]
```diff
@@ -1,14 +1,25 @@
 import React, { Component } from 'react';
 import { NavLink } from 'react-router-dom';
+import Modal from 'react-modal';
 import LessonBox from './LessonBox/LessonBox';
 import AddLessonBox from './AddLessonBox/AddLessonBox';
+import AddLessonModal from './Modal/AddLessonModal/AddLessonModal';
```
Comment (bchee) [inline client/src/components/Lessons/Lessons.jsx]: No need for two folders either name the folder Modal or AddLessonModal. Keep the file name the same

File: client/src/components/Lessons/Lessons.css [hunk 5]
```diff
@@ -25,3 +25,31 @@
...
+
+button:focus {
+  outline: 0;
+}
+
+.lessonbox:focus {
+  outline: 0;
+}
+
+.Modal {
```
Comment (bchee) [inline client/src/components/Lessons/Lessons.css]: there are many Modals so if others use this class name too you are gonna have an issue. I would either name it something more specific or make sure everyone else isn't using .Modal

File: client/src/components/Lessons/Lessons.css [hunk 6]
```diff
@@ -25,3 +25,31 @@
...
+  top: 50%;
+  left: 50%;
+  transform: translate(-50%, -50%);
+}
+
+.Modal:focus {
+  outline: none;
+}
+
+.Overlay {
```
Comment (bchee) [inline client/src/components/Lessons/Lessons.css]: Same goes for Overlay

**Comment by bchee** [comment]
Feel free to merge when you fix those comments.

File: client/src/components/Lessons/Lessons.css (hunk 6 above)
Comment (rrrliu) [reply client/src/components/Lessons/Lessons.css]: fixed, now named .LessonOverlay

File: client/src/components/Lessons/Lessons.css (hunk 5 above)
Comment (rrrliu) [reply client/src/components/Lessons/Lessons.css]: fixed, now named .LessonModal

File: client/src/components/Lessons/Lessons.jsx (hunk 4 above)
Comment (rrrliu) [reply client/src/components/Lessons/Lessons.jsx]: fixed, AddLessonModal folder removed

**Pull request merged**